### PR TITLE
Pubsub jobs inherit annotations defined on Prowjobs

### PR DESCRIPTION
**Pull request by chaodaiG**
This was missed from previous implementation

/cc @cjwagner @listx

### CODE REVIEW

File: prow/pubsub/subscriber/subscriber.go [hunk 1]
```diff
@@ -169,37 +169,37 @@ func (peh *periodicJobHandler) getProwJobSpec(cfg prowCfgClient, pc *config.InRe
 		}
 	}
 	if periodicJob == nil {
-		return nil, nil, fmt.Errorf("failed to find associated periodic job %q", pe.Name)
+		return nil, nil, nil, fmt.Errorf("failed to find associated periodic job %q", pe.Name)
```
Comment (listx) [inline prow/pubsub/subscriber/subscriber.go:172]: Nit: At some point we should stop adding more return values here and just return a struct.

File: prow/pubsub/subscriber/subscriber.go [hunk 2]
```diff
@@ -150,13 +150,13 @@ func (m *pubSubMessage) nack() {
 
 // jobHandler handles job type specific logic
 type jobHandler interface {
-	getProwJobSpec(cfg prowCfgClient, pc *config.InRepoConfigCacheHandler, pe ProwJobEvent) (*v1.ProwJobSpec, map[string]string, error)
+	getProwJobSpec(cfg prowCfgClient, pc *config.InRepoConfigCacheHandler, pe ProwJobEvent) (*v1.ProwJobSpec, map[string]string, map[string]string, error)
```
Comment (listx) [inline prow/pubsub/subscriber/subscriber.go:153]: Nit: Use some type alias here to name this string->string map into something else more meaningful.

**Review by listx — COMMENTED** [review]
/lgtm
/hold

**Comment by chaodaiG** [comment]
ack to both comments, will address in future PR
/unhold

**Pull request merged**